datalake-spark3/src/main/scala/bio/ferlab/datalake/spark3/implicits/ACMGImplicits.scala

```diff
@@ -46,6 +73,111 @@ object ACMGImplicits {
       }
     }
 
+    def getBS2(orphanet: DataFrame, frequencies: DataFrame): DataFrame = {
+
+      val map = Map(
+        df -> ("df", Array("symbol") ++ variantColumns),
+        orphanet -> ("orphanet", Array("gene_symbol", "average_age_of_onset", "type_of_inheritance")),
+        frequencies -> ("frequencies", Array("external_frequencies", "genes_symbol") ++ variantColumns)
+      )
+      validateRequiredColumns(map, "PM2")
+
+      val threshold = 4
+
+      val onsets = List(
+        "Adult",
+        "Elderly",
+        "All ages",
+        "No data available")
+
+      val is_dominant_inheritance = List(
+        "Autosomal dominant",
+        "X-linked dominant",
+        "Y-linked",
+        "Mitochondrial inheritance")
+
+      val orphanetDF = orphanet.select("gene_symbol", "average_age_of_onset", "type_of_inheritance")
+        .withColumn("is_adult_onset", inColArray("average_age_of_onset", onsets))
+        .filter(col("is_adult_onset") === false)
+        .withColumn("is_dominant", inColArray("type_of_inheritance", is_dominant_inheritance))
+        .select(
+          col("gene_symbol").as("symbol"),
+          col("is_dominant"))
+        .distinct()
+
+      val freqDF = frequencies
+        .select(
+          col("chromosome"),
+          col("start"),
+          col("end"),
+          col("reference"),
+          col("alternate"),
+          explode(col("genes_symbol")).as("symbol"),
+          col("external_frequencies.gnomad_genomes_3_1_1.ac").as("gnomad_ac"),
+          col("external_frequencies.gnomad_genomes_3_1_1.hom").as("gnomad_hom"))
```

Comment:
Maynbe we should prepare a table offline, and join on this instead of calculating this dataframe each time?